src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/DefaultDriverExceptionHandler.java

```diff
@@ -33,4 +42,136 @@ public RuntimeException handle(SchemaT schemaObject, ClosedConnectionException e
+    if (!allErrors.isEmpty()) {
+      List<Throwable> errors = allErrors.values().iterator().next();
+      if (errors != null && !errors.isEmpty()) {
+        Throwable error =
+            errors.stream()
+                .findAny()
+                .filter(
+                    t ->
+                        t instanceof AuthenticationException
+                            || t instanceof IllegalArgumentException
+                            || t instanceof NoNodeAvailableException
+                            || t instanceof DriverTimeoutException)
+                .orElse(null);
+
+        if (error == null) {
+          return exception;
+        }
+
+        return switch (error) {
+          case AuthenticationException e ->
+              // connect to OSS Cassandra throws AuthenticationException for invalid credentials
+              // TODO(Hazel): AuthException and INVALID_TOKEN?
+              AuthException.Code.INVALID_TOKEN.get(errVars(e));
+          case IllegalArgumentException e -> {
+            // AstraDB throws IllegalArgumentException for invalid token/credentials
+            if (e.getMessage().contains("AUTHENTICATION ERROR")
+                || e.getMessage()
+                    .contains("Provided username token and/or password are incorrect")) {
+              // TODO(Hazel): AuthException and INVALID_TOKEN?
+              yield AuthException.Code.INVALID_TOKEN.get(errVars(e));
+            } else {
+              yield exception;
+            }
+          }
+          case NoNodeAvailableException e -> handle(schemaObject, e);
+          case DriverTimeoutException e ->
+              // [data-api#1205] Need to map DriverTimeoutException as well
+              handle(schemaObject, e);
+          default -> exception;
+        };
+      }
+    }
+    return exception;
+  }
+
+  @Override
+  public RuntimeException handle(SchemaT schemaObject, NoNodeAvailableException exception) {
+    // TODO(Hazel): Aaron said NO_NODE_AVAILABLE is a bad code.
+    return DatabaseException.Code.NO_NODE_AVAILABLE.get(
+        errVars(schemaObject, map -> map.put(TemplateVars.ERROR_MESSAGE, exception.getMessage())));
+  }
+
+  @Override
+  public RuntimeException handle(SchemaT schemaObject, QueryValidationException exception) {
+    String message = exception.getMessage();
+    if (message.contains(
+            "If you want to execute this query despite the performance unpredictability, use ALLOW FILTERING")
+        || message.contains("ANN ordering by vector requires the column to be indexed")) {
+      // TODO(Hazel): Original code is NO_INDEX_ERROR, I think we need to change but am not sure
+      // what to change
+      return exception;
+    }
+    if (message.contains("vector<float,")) {
+      // It is tricky to find the actual vector dimension from the message, include as-is
+      // TODO(Hazel): the code VECTOR_SIZE_MISMATCH was added recently, should we keep using it?
```

Comment:
I think that'd be good because it is very common failure, from ops/metrics -- we do not want to lose this information wrt metrics (leads to unnecessary investigation for common failures).